File: Server/src/main/java/bgu/spl171/net/impl/TFTP/Commands/LoginRequest.java
package bgu.spl171.net.impl.TFTP.Commands;

import bgu.spl171.net.impl.TFTP.Commands.Responses.Response;

public class LoginRequest implements Command {
    String username;

    public String getUsername() {
        return username;
    }

    public LoginRequest(String username) {
        this.username = username;
    }

    @Override
    public short getOpCode() {
        return OpCode.Login.getValue();
    }

}

File: java to c++/LoginRequest.cpp
#include "LoginRequest.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						using Response = bgu::spl171::net::impl::TFTP::Commands::Responses::Response;

						std::wstring LoginRequest::getUsername()
						{
							return username;
						}

						LoginRequest::LoginRequest(const std::wstring &username)
						{
							this->username = username;
						}

						short LoginRequest::getOpCode()
						{
							return OpCode::Login::getValue();
						}
					}
				}
			}
		}
	}
}
